### PR TITLE
[docs]Remove any references encouraging users to write their token directly in their bot code. 

### PR DESCRIPTION
**Pull request by timmypidashev**
## Summary

PR for issue: #2518
Removes any references encouraging users to write their token directly in their bot code. 

```python
with open('token.txt') as f:
      token = f.read()

client.run(token) #or bot when in context
```

## Checklist

- [ ] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [x] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [x] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by lgaan** [comment]
I don’t think this is really needed.

The examples are there to provide a basic, one-file approach to basic concepts in the library; nowhere in the examples folder is a reference to loading tokens from separate files nor does this file exist leading to potentially more confusion.

Of course my say isn’t final, but this is just what I think.

**Comment by JelNiSlaw** [comment]
I think reading an environment variable would be a better solution

**Comment by timothypidashev** [comment]
I dont exactly think this is needed as well, but im just doing my part to clean up the issues page

**Comment by Phxntxm** [comment]
> I think reading an environment variable would be a better solution

I do not, I've seen far too many times where people come from these service that use `.env` file as a standard to import env variables. They then think that's normal, they have to do nothing and the file is magically loaded which is NOT the case in normal environments. I would prefer something that is an example to do something like this, reading from a file, where it's more direct what is happening.

Using environment variables then means you have to teach the user how to actually *create* environment variables, expanding the scope of this. Putting some text in a file is something you do not have to teach

**Comment by apple502j** [comment]
I'd say we should use `.env` and mention `python-dotenv`.

**Comment by Phxntxm** [comment]
> I'd say we should use `.env` and mention `python-dotenv`.

I VERY STRONGLY disagree with this, due to the first part I mentioned in my first reply. I have seen way too many people be bit by this from heroku/replit to support this

**Comment by lgaan** [comment]
> I'd say we should use `.env` and mention `python-dotenv`.

Back to my previous point, examples are meant to not rely on external packages.

**Comment by timothypidashev** [comment]
> Using environment variables then means you have to teach the user how to actually _create_ environment variables, expanding the scope of this. Putting some text in a file is something you do not have to teach

I agree with @Phxntxm. Enlarging the scope of this is the worst path to take in this situation. We simply don't want to promote hard-coding tokens into files that will most likely make it onto a repository later

File: examples/views/link.py [hunk 1]
```diff
@@ -36,4 +38,4 @@ async def google(ctx: commands.Context, *, query: str):
     await ctx.send(f'Google Result for: `{query}`', view=Google(query))
 
 
-bot.run('token')
+bot.run(token)
```
Comment (ioistired) [inline examples/views/link.py:41]: mising newline at EOF

File: examples/views/persistent.py [hunk 2]
```diff
@@ -60,4 +62,4 @@ async def prepare(ctx: commands.Context):
     await ctx.send("What's your favourite colour?", view=PersistentView())
 
 
-bot.run('token')
+bot.run(token)
```
Comment (ioistired) [inline examples/views/persistent.py:65]: missing a newline at EOF here too

**Review by ioistired — CHANGES_REQUESTED** [review]

File: examples/views/link.py (hunk 1 above)
Comment (timothypidashev) [reply examples/views/link.py]: will push in a sec

File: examples/views/persistent.py (hunk 2 above)
Comment (timothypidashev) [reply examples/views/persistent.py]: done

**Comment by ImportTaste** [comment]
If this isn't approved, at the very least, there should be a comment in the example files warning users never to share any file that has their token inside of it right above the line where they put their token in.

**Comment by timothypidashev** [comment]
> If this isn't approved, at the very least, there should be a comment in the example files warning users never to share any file that has their token inside of it right above the line where they put their token in.

agree

**Comment by Hyperclaw79** [comment]
Discord itself warns us while creating the token....I personally don't think there's a need to again tell people to not hard-code the token. It's something that should be pretty obvious....

**Comment by oliver-ni** [comment]
> Discord itself warns us while creating the token....I personally don't think there's a need to again tell people to not hard-code the token. It's something that should be pretty obvious....

They don't, actually.

**Comment by Hyperclaw79** [comment]
> They don't, actually.

oh....you're right. For some reason, I remember seeing a warning when we click on Copy/Reveal. Guess I confused it with some other app.
Either way, that warning should be provided by Discord itself, rather than a library.

**Comment by timothypidashev** [comment]
> Either way, that warning should be provided by Discord itself, rather than a library.
I agree Discord itself should provide a warning, but discord.py is not out of the picture either, and implying hard-coding tokens all throughout the docs and examples is not going to benefit anyone, in my humble opinion.

**Comment by timothypidashev** [comment]
The only inconsistency I can think of atm for this pr is the use of both `bot` and `client` when setting up the bot class. Should I make changes(switch all to client or bot), or keep as is?

**Comment by ioistired** [comment]
It's consistent with the fact that all instances of `commands.Bot` are called `bot` and all instances of `discord.Client` are called `client`, so keep it as is.

**Comment by timothypidashev** [comment]
> It's consistent with the fact that all instances of `commands.Bot` are called `bot` and all instances of `discord.Client` are called `client`, so keep it as is.

That is a good point, I will keep as is, thank you for pointing this out!

**Comment by timmypidashev** [comment]
Closing this pull request as it is stale and not relevant to current development.

**Pull request closed without merging**